### PR TITLE
Switch from Travis CI to GitHub Actions #585

### PR DESCRIPTION
**Pull request by ruspl-afed**
Added GitHub Actions config

Signed-off-by: Alexander Fedorov <alexander.fedorov@arsysop.ru>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      with:
+        java-version: 8
+    - name: Setup Maven
+      run: echo "MAVEN_OPTS='-Xmx2048m'" > ~/.mavenrc
+    - name: Cache maven repo
+      uses: actions/cache@v2
+      env:
+        cache-name: cache-maven-repo
+      with:
+        path: ~/.m2/repository
```
Comment (Flugtiger) [inline .github/workflows/ci.yml:37]: We might need to exclude BIRT artifacts from the cache to keep it at a reasonable size:
```suggestion
        path: |
        ~/.m2/repository
        !~/.m2/repository/org/eclipse/birt
        !~/.m2/repository/.cache/tycho
```

**Review by Flugtiger — CHANGES_REQUESTED** [review]

**Comment by Flugtiger** [comment]
Congrats to the first pr after the reboot 🥇 :-D

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ruspl-afed) [reply .github/workflows/ci.yml]: Done

**Comment by ruspl-afed** [comment]
Currently it took 20 mins to download dependencies

**Comment by ruspl-afed** [comment]
@Flugtiger I suggest to wait until GitHub Action will be enabled for this repo with https://bugs.eclipse.org/bugs/show_bug.cgi?id=571677 and then finalize this work

**Comment by ruspl-afed** [comment]
Dear @Flugtiger

@eclipsewebmaster told me 
> IIUC this is the first PR that defines a GitHub action, so it has to be merged first, before GitHub actions can be triggered on subsequent PRs.

https://bugs.eclipse.org/bugs/show_bug.cgi?id=571677#c1

So, I suggest to merge this intermediate PR to have GitHub Actions activated and then complete the work with another PR

**Comment by wimjongman** [comment]
Done

**Comment by ruspl-afed** [comment]
Thank you @wimjongman !

**Pull request merged**